### PR TITLE
feat(Vertica): Add Vertica dialect

### PR DESCRIPTION
**Pull request by Rajasekhar-Vuppala**
We are trying to add vertica dialect as we need this for datahub column level lineage. In this PR, We just added a base dialect and added few datatypes that were not there previously. upcoming PR's we may add more functionality.

We created Vertica.py we defined a class Vertica that inherits from the Dialect class in the sqlglot library. This class helps translate SQL expressions to Vertica-specific syntax. We imported tokenizer class to handle vertica specific quoting and identifiers , added keywords which extends default keywords by mapping vertica datatypes to their corresponding token type values.
generator class extend from sqlglot generator class to translate vertica specifc representations and added type_mapping dictionary that maps sqlglot datatypes to vertica datatypes.

Here is the [link](https://docs.vertica.com/24.1.x/en/sql-reference/data-types/) for Vertica Data types that we added.

Added test cases for all the datatypes that we mentioned above.

### CODE REVIEW

File: sqlglot/dialects/vertica.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class Vertica(Dialect):
+    class Tokenizer(tokens.Tokenizer):
+        QUOTES = ["'", '"']
+        IDENTIFIERS = ["`"]
+
+        KEYWORDS = {
+            **Tokenizer.KEYWORDS,
+            "MONEY": TokenType.MONEY,
+            "LONGVARBINARY": TokenType.VARBINARY,
+            "INTERVAL DAY TO SECOND": TokenType.INTERVAL,
```
[the remark this answers is not in the document]
Comment (Rajasekhar-Vuppala) [reply sqlglot/dialects/vertica.py:15]: Interval Day to second is Interval kind of datatype. You can find about this [here](https://docs.vertica.com/24.1.x/en/sql-reference/data-types/datetime-data-types/interval/)

File: sqlglot/dialects/vertica.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class Vertica(Dialect):
+    class Tokenizer(tokens.Tokenizer):
+        QUOTES = ["'", '"']
+        IDENTIFIERS = ["`"]
+
+        KEYWORDS = {
+            **Tokenizer.KEYWORDS,
+            "MONEY": TokenType.MONEY,
```
[the remark this answers is not in the document]
Comment (Rajasekhar-Vuppala) [reply sqlglot/dialects/vertica.py:13]: Can you please help us to understand where it is duplicated, in postgres and tsql?
So you meant we need to inherit from postgres or tsql ?

File: sqlglot/dialects/vertica.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            "LONGVARBINARY": TokenType.VARBINARY,
```
[the remark this answers is not in the document]
Comment (Rajasekhar-Vuppala) [reply sqlglot/dialects/vertica.py:14]: Yes, Varbinary and longvarbinary are two types as Long varbinary is not available we just thought and added here.
Please help us to understand what changes we need to do here

File: sqlglot/dialects/vertica.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/dialects/vertica.py:15]: Sorry, I'm not sure I understand the rationale here, can you elaborate further? This line essentially collapses the keyword sequence `INTERVAL DAY TO SECOND` into a single `INTERVAL` token. Why is this the case?

File: sqlglot/dialects/vertica.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/dialects/vertica.py:13]: The `KEYWORDS` entry `"MONEY": TokenType.MONEY` exists in both Postgres and T-SQL. My suggestion was to move this to the base tokenizer class in order to reduce this duplication.

File: sqlglot/dialects/vertica.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (georgesittas) [reply sqlglot/dialects/vertica.py:14]: > as Long varbinary is not available we just thought and added here.

What do you mean?

File: sqlglot/dialects/vertica.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            "REFRESH": TokenType.REFRESH,
+            "TEMP": TokenType.TEMPORARY,
+        }
+
+    class Generator(generator.Generator):
+        TYPE_MAPPING = {
+            exp.DataType.Type.MONEY: "MONEY",
+            exp.DataType.Type.VARBINARY: "LONGVARBINARY",
+            exp.DataType.Type.INTERVAL: "INTERVAL DAY TO SECOND",
+            exp.DataType.Type.INTERVAL: "INTERVAL YEAR TO MONTH",
```
Comment (georgesittas) [inline sqlglot/dialects/vertica.py:22]: Why is `INTERVAL` contained twice in this mapping?

File: sqlglot/dialects/vertica.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+        KEYWORDS = {
+            **Tokenizer.KEYWORDS,
+            "REFRESH": TokenType.REFRESH,
+            "TEMP": TokenType.TEMPORARY,
+        }
+
+    class Generator(generator.Generator):
+        TYPE_MAPPING = {
+            exp.DataType.Type.MONEY: "MONEY",
```
Comment (georgesittas) [inline sqlglot/dialects/vertica.py:19]: This is redundant

File: sqlglot/dialects/vertica.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+class Vertica(Dialect):
+    class Tokenizer(tokens.Tokenizer):
+        QUOTES = ["'", '"']
+        IDENTIFIERS = ["`"]
+
+        KEYWORDS = {
+            **Tokenizer.KEYWORDS,
+            "REFRESH": TokenType.REFRESH,
+            "TEMP": TokenType.TEMPORARY,
```
Comment (georgesittas) [inline sqlglot/dialects/vertica.py:14]: These don't show up in any test

**Review by georgesittas — COMMENTED** [review]
Hey @Rajasekhar-Vuppala, appreciate the interest to add Vertica, but this PR lacks proper testing and some changes are incorrect, as I mentioned in my comments.

I'd suggest studying the codebase a bit more before contributing, eg. to understand how the generator and parser classes actually work & how the various mappings are used.

In general, adding a dialect introduces new logic that needs to be maintained as the project grows, so the implementation needs to be both correct and also live up to the standards of the codebase.

I'll go ahead and close this PR. We'll be happy to discuss any questions or issues that come up in Slack, so feel free to open a thread there if you need assistance.

**Pull request closed without merging**